### PR TITLE
fix(java): Use Junit.assert instead of Hamcrest to fix Native Image testing

### PR DESCRIPTION
**Pull request by mpeddada1**
This PR is needed to address failures related to Native Image compilation. 
Calling `mvn test -Dtest=com.google.cloud.spanner.connection.it.ITQueryOptionsTest -Pnative -DfailIfNoTests=false` locally results in the following error:

```
JUnit Vintage:ITQueryOptionsTest:verifiesQueryOptions
    MethodSource [className = 'com.google.cloud.spanner.connection.it.ITQueryOptionsTest', methodName = 'verifiesQueryOptions', methodParameterTypes = '']
    => java.lang.Error: Cannot determine correct type for matchesSafely() method.
       org.hamcrest.internal.ReflectiveTypeFinder.findExpectedType(ReflectiveTypeFinder.java:49)
       org.hamcrest.TypeSafeMatcher.<init>(TypeSafeMatcher.java:40)
       org.hamcrest.TypeSafeMatcher.<init>(TypeSafeMatcher.java:22)
       org.hamcrest.core.SubstringMatcher.<init>(SubstringMatcher.java:13)
       org.hamcrest.core.StringStartsWith.<init>(StringStartsWith.java:13)
       org.hamcrest.core.StringStartsWith.startsWith(StringStartsWith.java:38)
       org.hamcrest.CoreMatchers.startsWith(CoreMatchers.java:516)
       com.google.cloud.spanner.connection.SqlScriptVerifier.verifyExpectedException(SqlScriptVerifier.java:188)
       com.google.cloud.spanner.connection.AbstractSqlScriptVerifier.verifyStatement(AbstractSqlScriptVerifier.java:367)
       com.google.cloud.spanner.connection.AbstractSqlScriptVerifier.lambda$verifyStatementsInFile$0(AbstractSqlScriptVerifier.java:315)
       [...]
```

GraalVM isn't able to handle more complex assertions in Hamcrest such as `assertThat().startsWith()` and `assertThat().contains()`. The Truth framework and Junit, on the other hand, are compatible.

cc @ansh0l @meltsufin

### CODE REVIEW

**Review by suztomo — APPROVED** [review]

File: google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/SqlScriptVerifier.java [hunk 1]
```diff
@@ -16,10 +16,8 @@
 
 package com.google.cloud.spanner.connection;
 
-import static org.hamcrest.CoreMatchers.equalTo;
-import static org.hamcrest.CoreMatchers.is;
-import static org.hamcrest.CoreMatchers.startsWith;
-import static org.hamcrest.MatcherAssert.assertThat;
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth.assertWithMessage;
```
Comment (ansh0l) [inline google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/SqlScriptVerifier.java:20]: @mpeddada1 I'm trying to understand the import for Truth: https://github.com/googleapis/java-spanner/blob/aac438e02e841c6ee44362f5f319505d0264ceda/google-cloud-spanner/pom.xml, but can't find it. Can you point me to the right pom?

**Review by ansh0l — APPROVED** [review]
lgtm, Since this is a change of just the assertion library. Adding a do not merge label since we intend to merge all the native image related PRs in a single go.

**Comment by ansh0l** [comment]
Closing since native image changes have been released with https://github.com/googleapis/java-spanner/pull/1878

**Comment by ansh0l** [comment]
Closing since native image changes have been released with https://github.com/googleapis/java-spanner/pull/1878

**Pull request closed without merging**